Protocol.md

```diff
@@ -467,18 +467,32 @@ Frame Contents
     |                         Stream ID = 0                         |
     +-----------+-+-+-+-------------+-------------------------------+
     |Frame Type |0|0|R|    Flags    |
-    +-----------+-+-+-+-------------+-------------------------------+
-    |0|                  Last Received Position                     |
-    +                                                               +
+    +-----------+-+-+-+-------------+---------------+---------------+
+    |0|0|0|0|           Number of Entries           |               
+    +-------------------------------+---------------+---------------+
+    |0|                         Stream ID                           |
+    +-------------------------------+-------------------------------+
     |                                                               |
-    +---------------------------------------------------------------+
+    |                       Implied Position                        |
+    |                                                               |
+    +-------------------------------+-------------------------------+
+    |0|                         Stream ID                           |
+    +-------------------------------+-------------------------------+
+    |                                                               |
+    |                       Implied Position                        |
+    |                                                               |
+    +-------------------------------+-------------------------------+
+    |                              ...                              |
+    +-------------------------------+-------------------------------+
                                   Data
 ```
 
 * __Frame Type__: (6 bits) 0x03
 * __Flags__: (10 bits)
      * (__R__)espond with KEEPALIVE or not
-* __Last Received Position__: (63 bits = max value 2^63-1) Unsigned 63-bit long of Resume Last Received Position. Value MUST be > 0. (optional. Set to all 0s when not supported.)
+* __Number of Entries__: (24 bits = max value 2^20 = 1,048,576) (optional. Set to all 0s when not supported.)
+* __Stream ID__: (31 bits = max value 2^31-1 = 2,147,483,647) Unsigned 31-bit integer representing the resumable stream Identifier. (present if Number of Entries > 0)
+* __Implied Position__: (64 bits = max value 2^64) Unsigned 64-bit value of the individual Stream Resume Last Received Frame Position from the remote party. Value MUST be > 0. (present if Number of Entries > 0)
```

Comment:
Will defer to others, to my mind the limiting to 2^63 makes it way simpler to safely store in a Long.  2^64 is really awkward in Java.